vlasiator.cpp

```diff
@@ -985,16 +1001,36 @@ int main(int argn,char* args[]) {
       //TODO - add LB measure and do LB if it exceeds threshold
       if(((P::tstep % P::rebalanceInterval == 0 && P::tstep > P::tstep_min) || overrideRebalanceNow)) {
          logFile << "(LB): Start load balance, tstep = " << P::tstep << " t = " << P::t << endl << writeVerbose;
-         // Refinement includes LB
-         if (!dtIsChanged && P::adaptRefinement && P::tstep % (P::rebalanceInterval * P::refineMultiplier) == 0 && P::t > P::refineAfter) { 
+         if (refineNow || (!dtIsChanged && P::adaptRefinement && P::tstep % (P::rebalanceInterval * P::refineMultiplier) == 0 && P::t > P::refineAfter)) { 
```

Comment:
Managed to dig up the commit introducing that from the blame. Changing dt doesn't advance the timestep, so if this check wasn't here it would refine twice when timestep changes. Since dt change is checked after LB/refinement, refinement should occur before timestep changing just fine.

However I'm not entirely certain the check is necessary, since I'm pretty sure load balances aren't doubled either in that case (and we're inside the `if` for that).

Reply:
Thank you! Now with this, digging into the main loop I again learned new things and ended up looking at git blames from 11 years ago. All good. I guess we could get rid of some head-scratching if there was a comment to that effect here, but that doesn't need to be your job.